src/tlo/methods/healthsystem.py

```diff
@@ -2782,21 +2781,29 @@ def apply(self, population):
             self.module.bed_days.availability = self._parameters['beds_availability']
 
 
-class HealthSystemChangePriorityPolicy(RegularEvent, PopulationScopeEventMixin):
+class HealthSystemChangePriorityPolicyAndMode(RegularEvent, PopulationScopeEventMixin):
     """ This event exists to change the priority policy adopted by the
     HealthSystem at a given year.    """
 
     def __init__(self, module):
         super().__init__(module, frequency=DateOffset(years=100))
 
     def apply(self, population):
-        self.module.priority_policy = self.module.parameters["policy_name_post_switch"]
+
+        # Change mode_appt_constraints
         self.module.mode_appt_constraints = self.module.parameters["mode_appt_constraints_postSwitch"]
-        if self.module.priority_policy != "":
-            self.module.load_priority_policy(self.module.priority_policy)
+
+        # If policy has changed, update it
+        if self.module.parameters["policy_name"] != self.module.parameters["policy_name_post_switch"]:
+            self.module.priority_policy = self.module.parameters["policy_name_post_switch"]
+            if self.module.priority_policy != "":
+                self.module.load_priority_policy(self.module.priority_policy)
```

Comment:
if the new priority policy is "" is it right to do nothing? What if the initial policy with something different? Don't we always need to actively set the policy to something.

Also,  I think the handling of the special case of "" should be inside `load_priority_policy` rather than repeated in the two places where the call is made.

Reply:
Mhm yes I'd say if the policy is "" it is right to do nothing, because in that case the policy will be ignored anyway, so it should be consistent. Will move the check to the function.

Reply:
But what if we want:
Period 1: Prioritise RMNCH
Period 2: Natural policy (i.e. "").

Will that be achieved?

Reply:
On line 1294
1294         if self.priority_policy != "":
1295             # Look-up priority ranking of this treatment_ID in the policy adopted
1296             priority = self.enforce_priority_policy(hsi_event=hsi_event)

This ensures that we will only look up a policy if a policy is not "". So if in Period 2 we set a policy "", this means for that period the priority assigned to the HSIs will not be looked up via the "enforce_priority_policy" fnc, but rather it will default to whatever was set by the disease modules themselves.

Reply:
Hence it is fine for no policy to have been loaded in that period.